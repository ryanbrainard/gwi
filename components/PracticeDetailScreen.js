import React from "react";
import { View } from "react-native";
import CharacterGrid from "./CharacterGrid";

export default class PracticeDetailScreen extends React.Component {
  // TODO: how to do prop types on navigation params

  static navigationOptions = ({ navigation }) => ({
    title: navigation.getParam("charSet").name
  });

  render() {
    const { navigation } = this.props;
    const charSet = navigation.getParam("charSet");

    return (
      <View style={{ flex: 1 }}>
        <CharacterGrid chars={charSet.groups} />
      </View>
    );
  }
}
