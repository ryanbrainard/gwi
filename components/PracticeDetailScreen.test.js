import React from "react";
import PracticeDetailScreen from "./PracticeDetailScreen";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const navigation = {
    getParam: jest.fn(() => ({ characters: [] }))
  };

  const tree = renderer
    .create(<PracticeDetailScreen navigation={navigation} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
