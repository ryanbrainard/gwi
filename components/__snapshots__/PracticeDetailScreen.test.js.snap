// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      invertStickyHeaders={undefined}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={3}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
